scripts/import_from_ena_legacy.py

```diff
@@ -0,0 +1,849 @@
+            es_doc_specimen['breed'] = check_existsence(specimen_detail['organism'], 'breed')
+            specimens_list.append(es_doc_specimen)
+            species[specimen_detail['organism']['organism']['text']] = specimen_detail['organism']['organism']
+        if not specimens_list:  # no sepcimen found for the dataset, skip
+            logger.warning(f"Dataset {dataset_id} could not find any related specimen, skipped")
+            continue
+        es_doc_dataset['specimen'] = sorted(specimens_list, key=lambda k: k['biosampleId'])
+        es_doc_dataset['species'] = list(species.values())
+        file_arr = datasets['tmp'][dataset_id]['file'].values()
+        valid_files = list()
+        for file_entry in sorted(file_arr, key=lambda k: k['name']):
+            file_id = file_entry['fileId']
+            if file_id in indexed_files:
+                valid_files.append(file_entry)
+        es_doc_dataset['file'] = valid_files
+        es_doc_dataset['experiment'] = list(only_valid_exps.values())
+        es_doc_dataset['assayType'] = list(experiment_type.keys())
+        es_doc_dataset['tech'] = list(tech_type.keys())
+        es_doc_dataset['instrument'] = list(datasets['tmp'][dataset_id]['instrument'])
+        es_doc_dataset['archive'] = sorted(list(datasets['tmp'][dataset_id]['archive']))
+        body = json.dumps(es_doc_dataset)
+        utils.insert_into_es(es, es_index_prefix, 'dataset', dataset_id, body)
+    logger.info("finishing indexing datasets")
+
+def get_ena_data():
+    """
+    This function will fetch data from ena FAANG data portal ruead_run result
+    :return: json representation of data from ena
+    """
+    response = requests.get('https://www.ebi.ac.uk/ena/portal/api/search/?result=read_run&format=JSON&limit=0'
+                            '&dataPortal=faang&fields=all').json()
+    return response
+
+
+def get_all_specimen_ids(host, es_index_prefix):
+    """
+    This function return dict with all information from the corresponding specimens
+    :return: A dict with keys as BioSamples id and values as the data stored in ES
+    """
+    if not host.endswith(":9200"):
+        host = host + ":9200"
+    results = dict()
+    url = f'http://{host}/{es_index_prefix}_specimen/_search?size=100000'
+    response = requests.get(url).json()
+    for item in response['hits']['hits']:
+        results[item['_id']] = item['_source']
+    return results
+
+
+def get_known_errors():
+    """
+    This function will read file with association from study to biosample
+    :return: dictionary with study as a key and biosample as a values
+    """
+    known_errors = dict()
+    with open('ena_not_in_biosample.txt', 'r') as f:
+        for line in f:
+            line = line.rstrip()
+            study, biosample = line.split("\t")
+            known_errors.setdefault(study, {})
+            known_errors[study][biosample] = 1
+    return known_errors
+
+
+def check_existsence(data_to_check, field_to_check):
```

Comment:
1. Documentation should be added
2. We already have this function in import_from_ena script, so it should be extracted to misc module

Reply:
removed in 27a573a